### PR TITLE
Starting work on error detection.

### PR DESCRIPTION
**Pull request by ue71603**
not yet sophisticated.

TODO:
* Information about things in the past
* Partial pricing should be noted correctly
* Some problem with local tariffs in Bern (probably we didn't have the right part of NOVA used there)

### CODE REVIEW

**Comment by ue71603** [comment]
simple solution found for https://github.com/openTdataCH/ojp-nova/issues/35 not yet good. But I close it anyhow. Error messages come with OJP 2.0

Also: https://github.com/openTdataCH/ojp-nova/issues/33
and https://github.com/openTdataCH/ojp-nova/issues/31

**Comment by ue71603** [comment]
@skinkie  Can you check and merge, if ok....

File: map_nova_to_ojp.py [hunk 1]
```diff
@@ -31,8 +31,10 @@ def map_preis_auspraegung_to_trip_fare_result(preis_auspraegungen: List[PreisAus
                                                                    fare_product_name=preis_auspraegung.produkt_nummer,
                                                                    fare_authority_ref='NOVA',
                                                                    fare_authority_text='NOVA',
+                                                                   price=round(float(preis_auspraegung.preis.betrag)*(1.07),2),
```
Comment (skinkie) [inline map_nova_to_ojp.py:34]: Mmm... this really gets specific, but why not 1.077?

**Review by skinkie — APPROVED** [review]

File: map_nova_to_ojp.py [hunk 2]
```diff
@@ -31,8 +32,10 @@ def map_preis_auspraegung_to_trip_fare_result(preis_auspraegungen: List[PreisAus
                                                                    fare_product_name=preis_auspraegung.produkt_nummer,
                                                                    fare_authority_ref='NOVA',
                                                                    fare_authority_text='NOVA',
-                                                                   net_price=preis_auspraegung.preis.betrag,
+                                                                   price=preis_auspraegung.preis.betrag,
+                                                                   net_price=round(float(preis_auspraegung.preis.betrag)*(1.0-VATRATE/100),2),
```
Comment (skinkie) [inline map_nova_to_ojp.py:36]: Is this really what we have in OJP. The price without VAT?

File: map_ojp_to_nova.py [hunk 3]
```diff
@@ -64,19 +66,30 @@ def map_fare_request_to_nova_request(ojp: Ojp, age: int=30) -> PreisAuskunftServ
         segments = []
         leg_start = None
         leg_end = None
+        leg_nr=0
         for leg in legs:
-
+            leg_nr=leg_nr+1
```
Comment (skinkie) [inline map_ojp_to_nova.py:71]:
```suggestion
            leg_nr = leg_nr + 1
```

File: map_ojp_to_nova.py [hunk 4]
```diff
@@ -64,19 +66,30 @@ def map_fare_request_to_nova_request(ojp: Ojp, age: int=30) -> PreisAuskunftServ
...
+            if leg.continuous_leg is not None:
+                # We can't price continuous legs. This in many cases may be correct, but not for sharing.
+                continue
             if leg.timed_leg is None:
                 continue
-
+            # if the timed leg is an on demand bus -> also ignore
+            if leg.timed_leg.service.mode is None:
+                continue
+            if leg.timed_leg.service.mode.bus_submode=="demandResponsive":
```
Comment (skinkie) [inline map_ojp_to_nova.py:81]:
```suggestion
            if leg.timed_leg.service.mode.bus_submode == "demandResponsive":
```

File: map_ojp_to_nova.py [hunk 5]
```diff
@@ -64,19 +66,30 @@ def map_fare_request_to_nova_request(ojp: Ojp, age: int=30) -> PreisAuskunftServ
         segments = []
         leg_start = None
         leg_end = None
+        leg_nr=0
         for leg in legs:
-
+            leg_nr = leg_nr + 1
```
Comment (skinkie) [inline map_ojp_to_nova.py:71]: Where in the code are you reusing leg_nr?

File: server.py [hunk 6]
```diff
@@ -8,11 +8,13 @@
 
 from map_nova_to_ojp import test_nova_to_ojp
 from map_ojp_to_ojp import parse_ojp
+from support import error_response
 from ojp import Ojp, Ojpresponse, ServiceDelivery, ServiceDeliveryStructure, \
     OtherError, OjpfareDelivery
 from test_network_flow import test_nova_request_reply, call_ojp_2000
-
-from configuration import HTTP_HOST, HTTP_PORT, HTTPS, SSL_CERTFILE, SSL_KEYFILE, HTTP_SLUG
+from configuration import HTTP_HOST, HTTP_PORT, HTTPS, SSL_CERTFILE, SSL_KEYFILE, HTTP_SLUG,VATRATE
```
Comment (skinkie) [inline server.py:15]:
```suggestion
from configuration import HTTP_HOST, HTTP_PORT, HTTPS, SSL_CERTFILE, SSL_KEYFILE, HTTP_SLUG, VATRATE
```

**Pull request merged**